### PR TITLE
DOC: update wheel build / upload instructions

### PR DESCRIPTION
**Pull request by matthew-brett**
Update the instructions for the numpy Windows and manylinux wheels
builds, and uploading the wheels to pypi.

### CODE REVIEW

**Comment by rgommers** [comment]
LGTM, thanks Matthew.

File: doc/HOWTO_RELEASE.rst.txt [hunk 1]
```diff
@@ -406,8 +458,44 @@ For example::
 Update PyPi
```
Comment (ogrisel) [inline doc/HOWTO_RELEASE.rst.txt]: Casing: PyPI.

**Comment by matthew-brett** [comment]
I fixed the casing of PyPI according to Olivier's suggestion.

Added more on uploading via warehouse, which seems already to be more reliable than pypi.python.org .

**Comment by charris** [comment]
Hmm, I notice there is no current mention of twine for uploading the source files. Not part of this PR, though.

**Comment by charris** [comment]
Thanks Matthew.

**Comment by charris** [comment]
@matthew-brett Looks like it needs an update for `manylinux1`, which seems to show up on `http://wheels.scipy.org/` these days when `numpy-wheels` is triggered.

**Comment by matthew-brett** [comment]
Chuck - yes - sorry - I was refactoring the build procedure as you were doing the release, that must have been distracting.

New procedure is more streamlined - all wheels get built from the same repo, and get uploaded to the same Rackspace container - PR at #7785 .

**Pull request merged**